### PR TITLE
feat/add myOkr, allOkr, styles Okr, State with data. Task 34 

### PR DESCRIPTION
**Pull request by twistedomelette**

### CODE REVIEW

File: frontend/src/components/okrs/components/tabs-okr/two-tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    const User = useSelector((state: RootState) => state.okr.User);
+    const OKR = useSelector((state: RootState) => state.okr.OKR);
+    const Objective = useSelector((state: RootState) => state.okr.Objective);
+    const KeyResult = useSelector((state: RootState) => state.okr.KeyResult);
+
+    return (
+      <Tabs
+        id="controlled-tab-example"
+        activeKey={key}
+        onSelect={(k:any):any => setKey(k)}
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/okrs/components/tabs-okr/two-tabs.tsx]: do not use any

File: frontend/src/components/okrs/components/tabs-okr/two-tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        className="mb-3"
+      >
+        <Tab eventKey="my-OKR" title="My OKR">
+            <div className="OKR-page">
+            {OKR.map((MyOKR: any) => {
+                const ArrayofObjectives = Objective.filter((el:any) => el.okrId === MyOKR.id);
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/okrs/components/tabs-okr/two-tabs.tsx]: do not use any

File: frontend/src/components/okrs/components/tabs-okr/two-tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                const NumOfObjectives = ArrayofObjectives.length;
+                let NumOfKeyResults = 0;
+                for (const obj of ArrayofObjectives)
+                    NumOfKeyResults += KeyResult.filter((el:any) => el.ObjectiveId === obj.id).length;
+                return <OrkItem key={MyOKR.id} name={MyOKR.name} endDate={MyOKR.endDate}
+                startDate={MyOKR.startDate} NumOfObjectives={NumOfObjectives} NumOfKeyResults={NumOfKeyResults}/>;
+            })}
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/okrs/components/tabs-okr/two-tabs.tsx]: logic is duplicated, reuse

File: frontend/src/store/okr/slice.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+type State = {
+    User: any,
+    // {id: number, email:string, password: number, firstname:string, lastname:string,
+    // companyId: number, mentorId: number, domenLevelId: number},
+    OKR: any,
+    // {id: number, name:string, startDate:string, endDate:string, userId:number}[],
+    Objective: any,
+    // {id: number, name:string, skillObjectiveId:number, okrId:number, result:number}[],
+    KeyResult: any,
+    // {id: number, name:string, ObjectiveId:number}[]
```
Comment (aleksandr-kuharenko) [inline frontend/src/store/okr/slice.ts]: Use interfaces

File: frontend/src/components/okrs/styles.scss [hunk 5]
```diff
@@ -0,0 +1,48 @@
+.okr-container {
+    margin-left: 15px;
```
Comment (nikita-remeslov) [inline frontend/src/components/okrs/styles.scss]: ms-3

File: frontend/src/components/okrs/styles.scss [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    margin-bottom: 15px;
```
Comment (nikita-remeslov) [inline frontend/src/components/okrs/styles.scss]: mb-3

File: frontend/src/components/okrs/styles.scss [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    width: 285px;
+    height: 175px;
+    border: 2px solid #ff9900;
+    border-radius: 10px;
+    padding: 8px 15px;
```
Comment (nikita-remeslov) [inline frontend/src/components/okrs/styles.scss]: py-2 px-3

File: frontend/src/components/okrs/styles.scss [hunk 8]
```diff
@@ -0,0 +1,48 @@
+.okr-container {
+    margin-left: 15px;
+    margin-bottom: 15px;
+    width: 285px;
+    height: 175px;
+    border: 2px solid #ff9900;
+    border-radius: 10px;
```
Comment (nikita-remeslov) [inline frontend/src/components/okrs/styles.scss]: rounded

File: frontend/src/components/okrs/styles.scss [hunk 9]
```diff
@@ -0,0 +1,48 @@
+.okr-container {
+    margin-left: 15px;
+    margin-bottom: 15px;
+    width: 285px;
+    height: 175px;
+    border: 2px solid #ff9900;
```
Comment (nikita-remeslov) [inline frontend/src/components/okrs/styles.scss]: try to use `border` class

File: frontend/src/components/okrs/styles.scss [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    border-radius: 10px;
+    padding: 8px 15px;
+    display: grid;
+    grid-template-columns: 1fr, 1fr, 30px;
+    grid-template-rows: 1fr, 1fr, 100px;
+    grid-template-areas:
+            "n n re"
```
Comment (nikita-remeslov) [inline frontend/src/components/okrs/styles.scss]: If you come back to this code in a month, I don't think you'll remember that grid area name.

File: frontend/src/components/okrs/styles.scss [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+            "t t t";
+}
+
+.OKR-name {
+    grid-area: n;
+}
+
+.reached {
+    grid-area: re;
+    align-self: start;
```
Comment (nikita-remeslov) [inline frontend/src/components/okrs/styles.scss]: align-self-start

File: frontend/src/components/okrs/styles.scss [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+.timestamp {
+    grid-area: t;
+    color: silver;
+}
+
+.OKR-page {
+    display: flex;
+    flex-direction: row;
```
Comment (nikita-remeslov) [inline frontend/src/components/okrs/styles.scss]: d-flex flex-row flex-wrap

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Comment by nikita-remeslov** [comment]
Also fix conflicts

**Comment by nikita-remeslov** [comment]
Conflcts

File: frontend/src/store/okr/slice.ts [hunk 13]
```diff
@@ -0,0 +1,133 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { ReducerName } from 'common/enums/app/reducer-name.enum';
+import { IUser, IOkr, IObjective, IKeyResult } from '../common/interfaces';
+
+type State = {
+    User: IUser,
```
Comment (nikita-remeslov) [inline frontend/src/store/okr/slice.ts]: `user`, use lowercase

File: frontend/src/store/common/interfaces.ts [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    name: string,
+    skillObjectiveId: number,
+    okrId: number,
+    result: number
+}
+
+export interface IKeyResult {
+    id: number,
+    name: string,
+    ObjectiveId: number
```
Comment (nikita-remeslov) [inline frontend/src/store/common/interfaces.ts]: lowercase

File: frontend/src/components/okr/workspace-item.tsx [hunk 15]
```diff
@@ -0,0 +1,39 @@
+import './styles.scss';
+import calendar from 'assets/img/okr-items/calendar-orange.png';
+import pencil from 'assets/img/okr-items/pencil.png';
+
+interface PropTypes {
+    name: string
+    startDate: string
+    endDate: string
+    NumOfObjectives: number
+    NumOfKeyResults: number
```
Comment (nikita-remeslov) [inline frontend/src/components/okr/workspace-item.tsx]: resultCount

File: frontend/src/components/okr/workspace-item.tsx [hunk 16]
```diff
@@ -0,0 +1,39 @@
+import './styles.scss';
+import calendar from 'assets/img/okr-items/calendar-orange.png';
+import pencil from 'assets/img/okr-items/pencil.png';
+
+interface PropTypes {
+    name: string
+    startDate: string
+    endDate: string
+    NumOfObjectives: number
```
Comment (nikita-remeslov) [inline frontend/src/components/okr/workspace-item.tsx]: objectiveCount

File: frontend/src/components/okr/two-tabs.tsx [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+
+function ControlledTabs():React.ReactElement {
+    const [key, setKey] = useState('my-OKR');
+    const user = useSelector((state: RootState) => state.okr.User);
+    const okr = useSelector((state: RootState) => state.okr.OKR);
+    const myOkr = useSelector((state: RootState) => state.okr.OKR.filter(okr => okr.userId === user.id));
+    const objective = useSelector((state: RootState) => state.okr.Objective);
+    const keyResult = useSelector((state: RootState) => state.okr.KeyResult);
+
+    const OkrList: React.FC<PropTypes> = (props) => (
```
Comment (nikita-remeslov) [inline frontend/src/components/okr/two-tabs.tsx]: 1 component === 1 file

Also, use can get collection with useSelector and pass it to the OkrList.

File: frontend/src/components/okr/common/interfaces.ts [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+    name: string,
+    skillObjectiveId: number,
+    okrId: number,
+    result: number
+}
+
+export interface IKeyResult {
+    id: number,
+    name: string,
+    ObjectiveId: number
```
Comment (nikita-remeslov) [inline frontend/src/components/okr/common/interfaces.ts]: lowercase

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]
Remove interface duplication. Overall looks good

**Review by aleksandr-kuharenko — APPROVED** [review]

**Review by deimonbody — APPROVED** [review]

**Comment by twistedomelette** [comment]
Added okr in folder components/ork and change store

**Pull request merged**